modules/features2d/src/blobdetector.cpp

```diff
@@ -325,13 +325,19 @@ void SimpleBlobDetectorImpl::detect(InputArray image, std::vector<cv::KeyPoint>&
 
         std::vector < Center > curCenters;
         findBlobs(grayscaleImage, binarizedImage, curCenters);
+        if(params.maxThreshold - params.minThreshold <= params.thresholdStep) {
+            // if the difference between min and max threshold is less than the threshold step
+            // we're only going to enter the loop once, so we need to add curCenters
+            // to ensure we still use minDistBetweenBlobs
+            centers.push_back(curCenters);
```

Comment:
This change is considered as invalid.
It produces invalid keypoint with mean point of centers ([code](https://github.com/opencv/opencv/blob/3.4.15/modules/features2d/src/blobdetector.cpp#L367-L374))

relates #20649